pkg/minishift/config/config_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var testDir string
+
+func TestNewInstanceConfig(t *testing.T) {
+	setup(t)
+	defer teardown()
+
+	expectedFilePath := filepath.Join(testDir, "fake-machine.json")
+	cfg, _ := NewInstanceConfig(expectedFilePath)
+
+	if cfg.FilePath != expectedFilePath {
+		t.Errorf("Expected path '%s'. Received '%s'", expectedFilePath, cfg.FilePath)
+	}
+}
+
+func TestJsonFileExists(t *testing.T) {
+	setup(t)
+	defer teardown()
+
+	filePath := filepath.Join(testDir, "fake-machine.json")
+	expectedOcPath := filepath.Join(testDir, "fakeOc")
+	cfg := &InstanceConfig{
+		FilePath: filePath,
+		OcPath:   expectedOcPath,
+	}
+
+	jsonData, _ := json.MarshalIndent(cfg, "", "\t")
+	ioutil.WriteFile(cfg.FilePath, jsonData, 0644)
+
+	newCfg, _ := NewInstanceConfig(filePath)
+	if newCfg.OcPath != expectedOcPath {
+		t.Errorf("Expected oc path '%s'. Received '%s'", expectedOcPath, cfg.OcPath)
+	}
+}
+
+func TestWriteInstanceConfig(t *testing.T) {
+	setup(t)
+	defer teardown()
+
+	path := filepath.Join(testDir, "fake-machine.json")
+	cfg, _ := NewInstanceConfig(path)
+
+	expectedOcPath := filepath.Join(testDir, "fakeOc")
+	cfg.OcPath = expectedOcPath
+	cfg.WriteInstanceConfig()
+
+	if cfg.OcPath != expectedOcPath {
```

Comment:
So how does this test that the write worked? You are setting a value on cfg.OcPath and then assert against it and assume that the write worked? That's not really true. To test the write works, you need to make a change, write it and then either create a new instance of InstanceConfig against the same path and make sure it has the updated value or you bypass InstanceConfig and read the file directly. Either way, this does not test what the test name implies.